solution/index.js

```diff
@@ -0,0 +1,455 @@
+        addTask(tableEl, newTaskText)
+
+    }
+
+    if (event.target.className === "api-button"){ // Handles API buttons
+        if (event.target.id === "save-btn"){
+            handleAnswer("save")
+        }
+        if (event.target.id === "load-btn"){
+            handleAnswer("load")
+        }
+    }
+
+    if (event.target.id === "switch"){ // Handles Dark Mode toggle switch
+        document.getElementsByTagName("body")[0].classList.toggle("toggled-background")
+
+        document.getElementsByTagName("h1")[0].classList.toggle("toggled-title")
+
+        document.getElementsByClassName("add-task-div")[0].classList.toggle("toggled-search")
+        document.getElementsByClassName("add-task-div")[1].classList.toggle("toggled-search")
+        document.getElementsByClassName("add-task-div")[2].classList.toggle("toggled-search")
+        
+
+    } 
+    
+  })
+
+document.addEventListener("contextmenu", event => { // Listener for right click
+    event.preventDefault() // Prevent opening option window
+    if (event.target.className === "task"){ // Handles delete task
+        updateLocalStorage(findTableNameByElement(event.target.parentNode.parentNode), event.target, findTaskIndex(event.target), true)
+        clearDom()
+        saveDomAfterLoad()     
+    }
+});
+
+document.addEventListener("dblclick", event => { // Handles double click events
+    if (event.target.className === "task"){
+        event.target.setAttribute("contenteditable", true) // Enable the edit
+        event.target.addEventListener("blur", () => {editTask(event.target)}) 
+    }
+    
+})
+
+let mousePositionElement // Get live element mouse position
+document.addEventListener("mouseover", event =>{ // Handles hover events
+    mousePositionElement = event.target
+})
+
+document.addEventListener("keydown", event =>{ // Handles keyboard press events
+    if (event.altKey === true && (event.key === "1" || event.key === "2" || event.key === "3")){ // Checks if Alt + 1/2/3 is pressed
+        if (mousePositionElement.className === "task"){ // Checks if pressed on the task
+            let table
+            if (event.key === "1"){
+                table = "todo"
+            }
+            if (event.key === "2"){
+                table = "in-progress"
+            }
+            if (event.key === "3"){
+                table = "done"
+            }
+            moveTask(table, mousePositionElement)
+        }
+    }
+})
+
+document.getElementById("search").addEventListener("keyup", searchTask) // Handles keyboard press release
+
+window.addEventListener('load', saveDomAfterLoad) // Handles loading page
+
+// Drag & Drop Handle
+let draggableTask = null // Current dragged task element
+let originalDraggableTaskTable = null // Current dragged task's table element before drag
+let originalDraggableTaskIndex = null // Current dragged task's index in list before drag
+
+document.addEventListener("dragstart", event =>{ // Handles starting drag
```

Comment:
It is not effective to use a listening event on the document - the event should be specific